### ISSUE TITLE
Provide an example for NewTokenCredential 

### ISSUE THREAD

**Issue by patnaikshekhar** [Azure/azure-storage-queue-go#13]
It would be very useful to see an example with the NewTokenCredential call. Is this used to authenticate using managed service identity? What should the value of Initial Token Value be?

Thanks!

**Comment by zezha-msft** [comment]
Thanks @patnaikshekhar for reaching out!

The TokenCredential can be used for all scenarios. Are you particularly interested in MSI?

We can certain add a sample. I've logged this work item.

**Comment by patnaikshekhar** [comment]
Yes a MSI example would be great. That is a pretty common use case. Thanks!

**Comment by VivekDevops** [comment]
I am also trying to use NewTokenCredential to access azure blob with MSI. 
@patnaikshekhar  where you able to achieve this? 
@ zezha-msft  It would great to see a example on this.

**Comment by zezha-msft** [comment]
Hi @VivekDevops, thanks for reaching out! We are about to make some major improvements in this area. I'll make sure that we include an appropriate sample this time.

**Comment by VivekDevops** [comment]
Hi @zezha-msft thanks for the efforts. In the meantime, can you please give some guidance here.  I am trying to use NewTokenCredential to access azure blob. 

I am able to generate the TokenCredential. However when I try to use the same to access the azblob, I am getting below errors.

===== RESPONSE ERROR (ServiceCode=) =====
Description=Authentication information is not given in the correct format. Check the value of Authorization header.
RequestId:de4bc8a4-a01e-0074-7259-27ab1d000000
Time:2020-05-11T06:00:20.0601875Z, Details:
   Code: InvalidAuthenticationInfo
   PUT https://xxxx.blob.core.windows.net/xxx?restype=container&timeout=61
   Authorization: REDACTED
   User-Agent: [Azure-Storage/0.7 (go1.14.2; linux)]
   X-Ms-Client-Request-Id: [xx]
   X-Ms-Version: [2018-11-09]
   --------------------------------------------------------------------------------
   **RESPONSE Status: 400 Authentication information is not given in the correct format. Check the value of Authorization header.**
   Content-Length: [298]
   Content-Type: [application/xml]
   Date: [Mon, 11 May 2020 06:00:19 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Request-Id: [de4bc8a4-a01e-0074-7259-27ab1d000000]


The same token works, when I use the rest api instead of the azblob go sdk

**Comment by zezha-msft** [comment]
Hi @VivekDevops, we have [a test here that uses OAuth](https://github.com/Azure/azure-storage-blob-go/blob/3efca72bd11c050222deab57e25ea90df03b9692/azblob/zt_test.go#L241), perhaps it could help you?

**Comment by Segflow** [comment]
Can we have an example how to use ManagedIdentity?
